Table S1: Overview of modalities

allyhawkins opened the pull request:
Addresses #14 
Stacked on #13 

This PR adds in the script to generate a table summarizing the total libraries per project and what type of libraries are included. We decided we would like to include a table that has one row per project/ diagnosis grouping combination. So if a project has diagnoses that all fall under the same diagnoses group then there is only one row for that project. If the project contains samples that are from different diagnoses groups then there will be multiple lines. This really only happens when we have `Non-cancerous` samples present in a project. 

For each project/ diagnosis group, I show the total number of libraries and then break that down into single-cell, single-nucleus, bulk, spatial, with CITE, and with cellhashing. Note that if a project contains CITE or cell hashing, the total libraries does not count the CITE/ cell hash libraries twice. To count the total, I pulled out the unique libraries in each project. So if you see that a project contains 60 single-cell and then 60 with CITE-seq, then that just means all 60 single-cell libraries have associated CITE-seq data. 

For bulk and spatial, the library ID is not the same between the single-cell/ single-nuc library and the bulk/spatial library. So although they may be from the same sample (this is not always the case), they will get counted as separate libraries when getting the total. 

Note I did stack this on #13 because I'm using the sample and library metadata from S3 that is downloaded in one of the scripts added in #13.  

One major thing to note is that I am showing the total number of libraries and NOT the total number of samples. I'm curious if others think we should show the total number of samples too?

allyhawkins commented:
I'm going to request @jaclyn-taroni to just take a look at the table to make sure this is what she expected, and then another scientist for review of the code.

jaclyn-taroni commented on tables/Supplemental-table-1-modality-overview.tsv:
```diff
@@ -0,0 +1,21 @@
+scpca_project_id	Diagnosis group	Total number of libraries	Single-cell	Single-nucleus	Bulk RNA	With CITE-seq	Spatial transcriptomics	With cell hashing	Diagnoses
```
My only comment is that I would move `Diagnoses` immediately after `Diagnosis group`

jaclyn-taroni reviewed (commented):
This mostly looks as I expected. I left one comment about moving a column!

sjspielman commented on scripts/table-s1-modality-summary.R:
```diff
@@ -0,0 +1,109 @@
+# read in groupings
+diagnosis_groupings_df <- readr::read_tsv(diagnosis_groupings_file)
+
+# read in library metadatq and filter to included projects
+library_metadata_df <- readr::read_tsv(library_metadata_file) |> 
+  dplyr::filter(scpca_project_id %in% project_whitelist)
+
+# read in sample metadatq and filter to included projects
```
metadatq -> metadata

sjspielman commented on scripts/table-s1-modality-summary.R:
```diff
@@ -0,0 +1,109 @@
+      seq_unit == "nucleus"  & !(stringr::str_detect(technology,"CITE")) & !(stringr::str_detect(technology, "cellhash")) ~ "Single-nucleus",
+      seq_unit == "bulk" ~ "Bulk RNA",
+      seq_unit == "spot" ~ "Spatial transcriptomics",
+      stringr::str_detect(technology,"CITE") ~ "With CITE-seq",
+      stringr::str_detect(technology,"cellhash") ~ "With cell hashing"
```
If you move these earlier, you can get rid of some the `!str_detect` code

sjspielman commented on scripts/table-s1-modality-summary.R:
```diff
@@ -0,0 +1,109 @@
+  tidyr::pivot_wider(names_from = modality, values_from = n) |> 
+  dplyr::mutate(
+    # replace all NA with 0
+    across(everything(), ~ tidyr::replace_na(.x,0))
```
```suggestion
    across(everything(), ~ tidyr::replace_na(.x, 0))
```

sjspielman commented on scripts/table-s1-modality-summary.R:
```diff
@@ -0,0 +1,109 @@
+
+# export table 
+readr::write_tsv(modality_counts_df, output_table_file)
+  
```
new line

sjspielman reviewed (commented):
This is _extremely not_ a full review, just a quick scroll through to get a sense of the code and small things I saw during said scroll.

sjspielman commented on scripts/table-s1-modality-summary.R:
```diff
@@ -0,0 +1,109 @@
+  dplyr::group_by(scpca_project_id, diagnosis_group) |> 
+  dplyr::mutate(diagnosis = paste(unique(diagnosis), collapse = ";")) |>
+  # make sure that diagnosis now gets retained throughout creation of the table
+  dplyr::group_by(scpca_project_id, diagnosis_group, diagnosis) |> 
+  dplyr::count(scpca_project_id, modality) |> 
```
You can just count here, which will also do (and then get rid of, which is good!) the grouping. The outcome looks the same to me in the end, unless I've missed something?

```suggestion
  dplyr::count(scpca_project_id, diagnosis_group, diagnosis, modality) |> 
```

sjspielman reviewed (commented):
Overall the result table looks correct to me, and this is mostly there! See inline comments..

> One major thing to note is that I am showing the total number of libraries and NOT the total number of samples. I'm curious if others think we should show the total number of samples too?

I do think this is worth showing because libraries:samples is not 1:1, for several reasons (multiplexing, and as you noted in the PR, "So although they may be from the same sample (this is not always the case), they will get counted as separate libraries when getting the total.")

People _love_ statistics like this (samples and libraries), and we're already in the supplemental so might as well have two side-by-side tables. I think the code here is small enough to just tack that extra code on without _needing_ a function, but I'm not entirely sure I'm right...

We'll also want to keep these items in mind for captions. Speaking of which, what/when is our plan to write those? Waiting for the paper itself, or should we get going on them alongside figure/table creation?

allyhawkins commented:
> People love statistics like this (samples and libraries), and we're already in the supplemental so might as well have two side-by-side tables. I think the code here is small enough to just tack that extra code on without needing a function, but I'm not entirely sure I'm right...

I'm good with adding a column that has total samples to this existing table. But I'm not sure we want to have a separate table. I'm on the fence on this so I'll add it in and go from there. 

> We'll also want to keep these items in mind for captions. Speaking of which, what/when is our plan to write those? Waiting for the paper itself, or should we get going on them alongside figure/table creation?

I was thinking figure captions/ legends would be separate.

allyhawkins commented on scripts/table-s1-modality-summary.R:
```diff
@@ -0,0 +1,109 @@
+    # make a new column that summarizes all modality info
+    # options are cell, nucleus, bulk, spatial, cite, or multiplexed
+    # the assigned values will eventually be the column names used in the output
+    modality = dplyr::case_when(
+      seq_unit == "bulk" ~ "Bulk RNA",
+      seq_unit == "spot" ~ "Spatial transcriptomics",
+      stringr::str_detect(technology,"CITE") ~ "With CITE-seq",
+      stringr::str_detect(technology,"cellhash") ~ "With cell hashing",
+      seq_unit == "cell" ~ "Single-cell",
+      seq_unit == "nucleus" ~ "Single-nucleus",
+    )
```
Yea I'm using the library metadata file that has one row per run.

sjspielman commented:
> I'm good with adding a column that has total samples to this existing table.

Seems easier!!

allyhawkins commented:
Okay I added in a column for total sample count and implemented your other suggestions @sjspielman. 

I think we should decide if we like just adding a column with the total samples, but then noting in the caption that the other columns show the total in libraries. Alternatively, we could create an entirely separate table where we show the breakdown of samples rather than libraries. I do think the separate table would be less confusing then showing one column with sample counts and all other columns with library counts. If we do that though I think we need another PR. I don't think the data wrangling will be exactly the same. Thoughts @jaclyn-taroni?

sjspielman commented on scripts/table-s1-modality-summary.R:
```diff
@@ -0,0 +1,109 @@
+# create a list of all the modality values
+# we will use this to total the number of libraries for a project across all modalities 
+# these will also correspond to the column names in the final table that contain the total for each modality 
+modality_columns <- c("Single-cell", "Single-nucleus", "Bulk RNA", "Spatial transcriptomics", "With CITE-seq", "With cell hashing")
```
Sticking with `CITE-seq`. See https://github.com/AlexsLemonade/scpca-paper-figures/pull/18#issuecomment-1883480435

sjspielman commented:
Hm, now I do wonder if it is confusing that the table modality columns show libraries. But, I think that confusion is easily addressed with little superscripts for those columns in the final table itself, with an associated footnote or caption section that says "these are library, not sample, counts."

allyhawkins commented:
> Hm, now I do wonder if it is confusing that the table modality columns show libraries, *given the additional # samples column. But, I think that confusion is easily addressed with little superscripts for those columns in the final table itself, with an associated footnote or caption section that says "these are library, not sample, counts."

I'm good with the idea of incorporating superscripts or some other indicator that they are libraries. What about adding (L) or (S) to the column titles to indicate which count is shown? Most of them will be L except for the total samples. Then we can describe the (L) and (S) in the caption?

sjspielman commented:
Yeah, I think either of those ideas are fine! The specific format (superscript vs indicator string) may end up depending on the journal, so we can maybe decide on the right approach once we finalize our submission plan?

allyhawkins commented:
For now, I just added the S and L indicators to the column names. If we need to change the formatting for the journal we can do that at a later time.
